### PR TITLE
dump: Simplify writeNode and use fewer goroutines

### PR DESCRIPTION
**Pull request by greatroar**
What does this PR change? What problem does it solve?
-----------------------------------------------------

This changes Dumper.writeNode to spawn loader goroutines as needed instead of as a pool. The code is shorter, fewer goroutines are spawned for small files, and crash dumps (also for unrelated errors) should be smaller.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------

No.

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added tests for all code changes.
- [ ] I have added documentation for relevant changes (in the manual).
- [ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

File: internal/dump/common.go [hunk 1]
```diff
@@ -104,75 +103,50 @@ func (d *Dumper) WriteNode(ctx context.Context, node *restic.Node) error {
...
+					return err
+				}
 			}
 		}
 		return nil
 	})
 
-	for i := uint(0); i < d.repo.Connections(); i++ {
+	// Start short-lived goroutines to load blobs.
+	// There will be at most 1+cap(blobs) calling LoadBlob at any moment.
```
Comment (greatroar) [inline internal/dump/common.go:125]: Note that the number of live goroutines can be higher, if the scheduler determines to leave them dangling just after ch <- blob, below. We can easily limit the number of goroutines to exactly cap(blobs) by sending ch before spawning. Would that be better?

File: internal/dump/common.go [hunk 2]
```diff
@@ -104,75 +103,50 @@ func (d *Dumper) WriteNode(ctx context.Context, node *restic.Node) error {
...
-		for result := range writerCh {
-			select {
-			case data := <-result.data:
-				if _, err := w.Write(data); err != nil {
-					return errors.Wrap(err, "Write")
-				}
-			case <-ctx.Done():
-				return ctx.Err()
-			}
+		select {
```
Comment (greatroar) [inline internal/dump/common.go:143]: This select isn't really necessary as LoadBlob and the writer goroutine respond to ctx being cancelled, but then we might do some unnecessary work.

**Review by MichaelEischer — APPROVED** [review]
LGTM. Thanks for the simplification.

File: internal/dump/common.go (hunk 1 above)
Comment (MichaelEischer) [reply internal/dump/common.go:125]: I don't think it matters, in the worst case the additional goroutine ends up blocked in the backend waiting for its turn.

**Pull request merged**